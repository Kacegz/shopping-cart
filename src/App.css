#navbar{
  height: 40px;
  background-color: antiquewhite;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px;
}
#main{
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

#footer{
  background-color: beige;
  justify-self: flex-end;
  display: flex;
  justify-content: center;
  align-items: center;
}