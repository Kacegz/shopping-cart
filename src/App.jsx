import { useState } from "react";
import styles from "./App.module.css";
import { Link, Outlet } from "react-router-dom";
import { useEffect } from "react";
import cartimg from "./assets/cart.svg";
import gem from "./assets/gem.svg";

function App() {
  const [cart, setCart] = useState([]);
  return (
    <>
      <div className={styles.navbar}>
        <div className={styles.flexgap}>
          <Link to="/" className={styles.flex}>
            <span className={styles.flex}>
              <img src={gem} className={styles.gem} alt="" srcSet="" />
              Home
            </span>
          </Link>
          <Link to="shop">
            <div>Shop</div>
          </Link>
        </div>
        <Link to="cart">
          <div className={styles.flex}>
            <img src={cartimg} className={styles.navimg} alt="" srcSet="" />
            {cart.length}
          </div>
        </Link>
      </div>
      <div className={styles.main}>
        <Outlet context={[cart, setCart]} />
      </div>
    </>
  );
}

export default App;
