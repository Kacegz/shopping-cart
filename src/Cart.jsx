const Cart = () => {
  return (
    <>
      <div id="main">
        <p>displaying x elements in cart</p>
      </div>
    </>
  );
};
export default Cart;
