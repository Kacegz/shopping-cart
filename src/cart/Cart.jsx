import { useOutletContext } from "react-router-dom";
import styles from "./Cart.module.css";

const Cart = () => {
  const [cart, setCart] = useOutletContext();
  function handleRemove(item) {
    const fixedCart = cart.filter((cartItem) => cartItem != item);
    setCart([...fixedCart]);
  }
  return (
    <>
      <div className={styles.main}>
        <div className={styles.itemList}>
          {cart.length > 0 ? (
            cart.map((item) => {
              return (
                <div className={styles.item} key={item.id}>
                  <p>{item.title}</p>

                  <div className={styles.spacing}>
                    <p>Q:{item.amount}</p>
                    <p>{item.amount * item.price}$</p>
                    <input
                      className={styles.removeButton}
                      type="button"
                      value="Remove"
                      onClick={() => {
                        handleRemove(item);
                      }}
                    />
                  </div>
                </div>
              );
            })
          ) : (
            <div className={styles.cartEmpty}>
              <h3>Cart is empty</h3>
            </div>
          )}
        </div>
        <div className={styles.total}>
          <h2>Summary</h2>
          <p>
            Total:
            {cart.reduce((sum, { price, amount }) => {
              return sum + price * amount;
            }, 0)}
            $
          </p>
          <input
            className={styles.checkoutButton}
            type="button"
            value="Checkout"
          />
        </div>
      </div>
    </>
  );
};
export default Cart;
