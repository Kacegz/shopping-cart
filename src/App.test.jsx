import { screen, render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import App from "./App";
import Cart from "./Cart";
import { BrowserRouter } from "react-router-dom";
describe("App", () => {
  it("Renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
