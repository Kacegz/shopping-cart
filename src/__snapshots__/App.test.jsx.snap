// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Renders correctly 1`] = `
<div>
  <div
    class="_navbar_6dc87e"
  >
    <div
      class="_flexgap_6dc87e"
    >
      <a
        class="_flex_6dc87e"
        href="/"
      >
        <span
          class="_flex_6dc87e"
        >
          <img
            alt=""
            class="_gem_6dc87e"
            src="/src/assets/gem.svg"
            srcset=""
          />
          Home
        </span>
      </a>
      <a
        href="/shop"
      >
        <div>
          Shop
        </div>
      </a>
    </div>
    <a
      href="/cart"
    >
      <div
        class="_flex_6dc87e"
      >
        <img
          alt=""
          class="_navimg_6dc87e"
          src="/src/assets/cart.svg"
          srcset=""
        />
        0
      </div>
    </a>
  </div>
  <div
    class="_main_6dc87e"
  />
</div>
`;
